Zend/zend_fibers.c

```diff
@@ -711,13 +712,15 @@ ZEND_METHOD(Fiber, getReturn)
 
 	fiber = (zend_fiber *) Z_OBJ_P(getThis());
 
-	if (fiber->status != ZEND_FIBER_STATUS_RETURNED) {
+	if (fiber->status != ZEND_FIBER_STATUS_DEAD || fiber->flags) {
```

Comment:
We can only get the return value after the fiber is dead, so the  `|| fiber->flags` here seem unnecessary, and it assumes that the flags is always used to indicate the error state, and we may add more other flags in the future.

Reply:
A fiber can be dead but have not returned a value (i.e., threw an exception). If flags are used for something other than error conditions then we'll have to revisit this, but it's fine for now.

Reply:
I mean
```C
if (fiber->status == ZEND_FIBER_STATUS_INIT) {
    ...
} else if (fiber->status == ZEND_FIBER_STATUS_DEAD) {
   if (fiber->flags ...) { ... }
} else {
   ...
}
```

It looks like you're being lazy.